kratos/tests/test_tensor_adaptors.py

```diff
@@ -0,0 +1,630 @@
+            self.assertEqual(node.Z, numpy_data[i, 2])
+
+    def test_NodePositionTensorAdaptor2(self):
+        node_position_ta = Kratos.TensorAdaptors.NodePositionTensorAdaptor(self.model_part.Nodes, Kratos.Configuration.Initial)
+        node_position_ta.CollectData()
+
+        numpy_data = node_position_ta.data
+        for i, node in enumerate(self.model_part.Nodes):
+            self.assertVectorAlmostEqual(numpy_data[i, :], [node.X0, node.Y0, node.Z0])
+
+        node_position_ta_1 = Kratos.TensorAdaptors.NodePositionTensorAdaptor(self.model_part.Nodes, Kratos.Configuration.Initial, [2])
+        node_position_ta_1.CollectData()
+        numpy_data_1 = node_position_ta_1.data
+        for i, node in enumerate(self.model_part.Nodes):
+            self.assertVectorAlmostEqual(numpy_data_1[i, :], [node.X0, node.Y0])
+
+        numpy_data_1[:] = numpy_data[:, 0:2] * 2
+        node_position_ta_1.StoreData()
+        for i, node in enumerate(self.model_part.Nodes):
+            self.assertEqual(node.X0, numpy_data[i, 0] * 2)
+            self.assertEqual(node.Y0, numpy_data[i, 1] * 2)
+            self.assertEqual(node.Z0, numpy_data[i, 2])
+
+        with self.assertRaises(RuntimeError):
+            Kratos.TensorAdaptors.NodePositionTensorAdaptor(Kratos.TensorAdaptors.VariableTensorAdaptor(self.model_part.Conditions, Kratos.VELOCITY), Kratos.Configuration.Initial)
+
+        with self.assertRaises(RuntimeError):
+            Kratos.TensorAdaptors.NodePositionTensorAdaptor(Kratos.TensorAdaptors.VariableTensorAdaptor(self.model_part.Elements, Kratos.VELOCITY), Kratos.Configuration.Initial)
+
+        with self.assertRaises(RuntimeError):
+            Kratos.TensorAdaptors.NodePositionTensorAdaptor(Kratos.TensorAdaptors.VariableTensorAdaptor(self.model_part.Properties, Kratos.VELOCITY), Kratos.Configuration.Initial)
+
+    def test_BaseTensorAdaptor(self):
+        var_ta = Kratos.TensorAdaptors.VariableTensorAdaptor(self.model_part.Nodes, Kratos.VELOCITY)
+        var_ta.Check()
+        var_ta.CollectData()
+
+        copied_ta = Kratos.TensorAdaptors.DoubleTensorAdaptor(var_ta, copy=True)
+        base_ta = Kratos.TensorAdaptors.DoubleTensorAdaptor(var_ta)
+        base_ta.data += 1.0
+
+        var_ta.StoreData()
+
+        for i, node in enumerate(self.model_part.Nodes):
+            self.assertEqual(node.GetValue(Kratos.VELOCITY_X), copied_ta.data[i, 0] + 1)
+            self.assertEqual(node.GetValue(Kratos.VELOCITY_Y), copied_ta.data[i, 1] + 1)
+            self.assertEqual(node.GetValue(Kratos.VELOCITY_Z), copied_ta.data[i, 2] + 1)
+
+        def check_entities(container_1, container_2):
+            self.assertEqual(len(container_1), len(container_2))
+            for entity_1, entity_2 in zip(container_1, container_2):
+                self.assertEqual(id(entity_1), id(entity_2))
+
+        check_entities(var_ta.GetContainer(), self.model_part.Nodes)
+        check_entities(base_ta.GetContainer(), self.model_part.Nodes)
```

Comment:
this doesn't check what you want to check.

I assume you want to make sure that the container storing pointers to nodes is not copied. That's a good idea, but not at all what this function is doing.